src/pal/src/loader/module.cpp

```diff
@@ -57,8 +57,10 @@ Module Name:
 #include <sys/types.h>
 #include <sys/mman.h>
 
-#if defined(__LINUX__)
+#if defined(__LINUX__) && defined(__GLIBC__)
 #include <gnu/lib-names.h>
+#else
+#define LIBC_SO "libc.so"
```

Comment:
Do you really want to define it for NetBSD? https://github.com/dotnet/coreclr/blob/49f39af14e6e2d4048f3c74db0c5164c0f4131b5/src/pal/src/loader/module.cpp#L1625